aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -461,6 +462,7 @@ Tensor scalar_tensor(Scalar s, const TensorOptions& options) {
     // In the future when we remove the overhead of device dispatch, we'll happily
     // revert this to following:
     //   auto result = at::empty({}, options);
+    at::tracer::impl::NoTracerDispatchMode tracer_guard;
     at::AutoNonVariableTypeMode non_var_type_mode(true);
```

Comment:
Delete AutoNonVariableTypeMode

Actually, TBH, I don't think the `NoTracerDispatchMode` is doing anything here either, because neither `empty_cpu` nor `fill_` should be hitting device dispatch

Reply:
Why did we introduce `AutoNonVariableTypeMode` here in the first place?
I agree empty_cpu / fill_ seems should be simple leaf functions, but I wasn't quite certain how reliable this assumption is (to be future-proof)